skbeam/core/roi.py

```diff
@@ -640,3 +641,40 @@ def box(shape, v_edges, h_edges=None, h_values=None, v_values=None):
             coords.append((h[0], v[0], h[1]-h[0], v[1] - v[0]))
 
     return rectangles(coords, v_values.shape)
+
+
+def lines(end_points, shape):
+    """
+    Parameters
+    ----------
+    end_points : iterable
+        coordinates of the starting point and the ending point of each
+        line: e.g., [(x1, y1, x2, y2), (x1, y1, x2, y2)]
+    shape : tuple
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. Order is (rr, cc).
+
+    Returns
+    -------
+    label_array : array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, corresponding to the order they are specified
+        in coords. Order is (rr, cc).
+
+    """
+    label_array = np.zeros(shape, dtype=np.int64)
+    l = 0
+    for points in end_points:
+        if len(points) == 4:
```

Comment:
Can you change this to be:

```python
for points in end_points:
    if len(points) != 4:
        raise ValueError(err_text)
    rr, cc = line(...)
    ...
```

I prefer to do this sort of testing by testing for the unexpected condition `len(points) != 4` instead of testing for the expected condition `len(points) == 4`.  I think it makes the code clearer if the expected code path is not wrapped inside of an if block.